### ISSUE TITLE
Have Infer default to using Enumerate

### ISSUE THREAD

**Issue by null-a** [probmods/webppl#629]

**Comment by null-a** [comment]
Once enumerate is the default method for `Infer` there will be cases where the params object is empty. Rather than expecting users to write `Infer({}, model)` we said we'd make `Infer(model)` work, and with that we may as well make `Infer(model, params)` work too.

I just realized that there's another thing we could do here -- we could move the model parameter into `params`. This addresses the specific problem raised by `Infer` having a default method, and I think it's a little nicer than having to write code to handle positional arguments in any order.

In practice it would look like this:

```js
// default to enumerate
Infer({model: function() {
  // do things
}})
```

```js
// some other method
Infer({method: 'MCMC', model: function() {
  // do things
}})
```

```js
// a method with lots of parameters
Infer({
  method: 'MCMC',
  kernel: 'HMC',
  samples: 1000,
  model: function() {
    // do things
  }
});
```

```js
// model as named function
Infer({method: 'MCMC', model: model})
```

This would be improved if we added started support some ES6 features:

```js
// short hand for functions in object literals
Infer({method: 'MCMC', model() {
  // do things
}})
```

```js
// short hand for properties in object literals
// when the model function is bound to a var named model
Infer({model, method: 'MCMC'}})
```

At first glance this seems as though it would be quite pleasant to use, and I'm pretty sure it would be simpler to implement and document. There's always a concern about breaking existing code, but if we wanted this, perhaps we could continue to support the existing signature as well, at least for the time been? Any thoughts?
